src/storage/MonitoringStore.ts

```diff
@@ -29,31 +31,44 @@ export class MonitoringStore<T extends ResourceStore = ResourceStore>
   }
 
   public async addResource(container: ResourceIdentifier, representation: Representation,
-    conditions?: Conditions): Promise<ResourceIdentifier> {
-    const identifier = await this.source.addResource(container, representation, conditions);
-    this.emitChanged([ container, identifier ]);
-    return identifier;
+    conditions?: Conditions): Promise<Record<string, RepresentationMetadata>> {
+    return this.emitChanged(await this.source.addResource(container, representation, conditions));
   }
 
   public async deleteResource(identifier: ResourceIdentifier,
-    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    conditions?: Conditions): Promise<Record<string, RepresentationMetadata>> {
     return this.emitChanged(await this.source.deleteResource(identifier, conditions));
   }
 
   public async setRepresentation(identifier: ResourceIdentifier, representation: Representation,
-    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    conditions?: Conditions): Promise<Record<string, RepresentationMetadata>> {
     return this.emitChanged(await this.source.setRepresentation(identifier, representation, conditions));
   }
 
   public async modifyResource(identifier: ResourceIdentifier, patch: Patch,
-    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    conditions?: Conditions): Promise<Record<string, RepresentationMetadata>> {
     return this.emitChanged(await this.source.modifyResource(identifier, patch, conditions));
   }
 
-  private emitChanged(identifiers: ResourceIdentifier[]): typeof identifiers {
-    for (const identifier of identifiers) {
-      this.emit('changed', identifier);
+  private emitChanged(changes: Record<string, RepresentationMetadata>): Record<string, RepresentationMetadata> {
+    for (const [ key, value ] of Object.entries(changes)) {
+      const activity = value.get(SOLID_AS.terms.Activity)?.value;
+      this.emit('changed', { path: key }, activity);
+      switch (activity) {
+        case AS.Create:
+          this.emit('created', { path: key });
+          break;
+        case AS.Delete:
+          this.emit('deleted', { path: key });
+          break;
+        case AS.Update:
+          this.emit('updated', { path: key });
+          break;
+        default:
+          break;
+      }
```

Comment:
Would define a mapping between the strings and terms above, so something like
```
const activityMap = {
  [AS.Create]: 'created',
  [AS.Delete]: 'deleted',
  [AS.Updated]: 'updated',
}
```
And then read the value from that map so the emit lines don't have to be duplicated 3 times.

Having said that, why do we use created/deleted/updated as event names here, and also Create/Delete/Update as values for the `changed` event above? Why not use AS.Create... as event names and as values for the `changed` event?  Seems like a safer way to make sure we always use the correct terms and don't accidentally have a typo somewhere. Also feels weird that there is a difference between the event names and the values for `changed`.

Does this perhaps have something to do with things that will be needed for the notifications spec?

Pinging @RubenVerborgh as he always cares about names.

Update after re-reading everything: I think this is what already happens as the value in the `changed` event right?  Since the value of the named node is used. But that is not what the changes to `documentation/resource-store.md` says which caused my confusion here.

Reply:
The `updated` and `update` difference was based on this comment: https://github.com/CommunitySolidServer/CommunitySolidServer/pull/1350#issuecomment-1167721968 .

This might have been a typo tho.
Do I changed all of them to `updateD` or `update` so I can use the AS namespace.

Reply:
Note that what I suggest is to use `AS.Create`, which is different from "Create". This is what you already do here when firing the `changed` event. This is not the string `"Create"`, but `"https://www.w3.org/ns/activitystreams#Create"`. Although the documentation currently says something different which caused my confusion.

So my question is, why not just use that full URL as well for the event name, instead of the current `"created"`, or even `"Create"`.

Reply:
Ah yes, I misunderstood. using `AS.Create` as the event name would indeed make more sense.
Will make those changes!

Reply:
> So my question is, why not just use that full URL as well for the event name, instead of the current `"created"`, or even `"Create"`.

Sounds good to me. We can use the full activity stream URLs as event name :+1: